internal/pkg/describe/webapp.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package describe
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+)
+
+// webAppURI represents the unique identifier to access a web application.
+type webAppURI struct {
+	DNSName string // The environment's subdomain if the application is served on HTTPS. Otherwise, the public load balancer's DNS.
+	Path    string // Empty if the application is served on HTTPS. Otherwise, the pattern used to match the application.
+}
+
+func (uri *webAppURI) String() string {
+	if uri.Path != "" {
+		return fmt.Sprintf("%s and path %s", color.HighlightResource("http://"+uri.DNSName), color.HighlightResource(uri.Path))
+	}
+	return color.HighlightResource("https://" + uri.DNSName)
+}
+
+// webAppDescriber retrieves information about a load balanced web application.
+type webAppDescriber struct {
+	*AppDescriber
+	stackDescribers map[string]stackDescriber
+}
+
+func newWebAppDescriber(parent *AppDescriber) *webAppDescriber {
+	return &webAppDescriber{
+		AppDescriber:    parent,
+		stackDescribers: make(map[string]stackDescriber),
```

Comment:
It seems like this map would only ever have 0 or 1 value in it. When would that not be the case? Should we just stash the client in an unexported field instead of this map and initialize it if it's uninitialized?

Reply:
Good point!! I changed things around especially in `describe.go`, so that the same "describer" can be re-used multiple times.